sidekick_core/lib/src/sidekick_context.dart

```diff
@@ -0,0 +1,160 @@
+import 'dart:io';
+
+import 'package:dcli/dcli.dart';
+import 'package:sidekick_core/sidekick_core.dart';
+import 'package:sidekick_core/sidekick_core.dart' as core;
+import 'package:sidekick_core/src/sidekick_package.dart';
+
+/// Environment variable containing the location of the shell `entryPoint`, when
+/// executing the sidekick CLI with the shell entrypoint
+///
+/// May be not set when debugging where the CLI is executed directly on the
+/// DartVM by calling `dart bin/main.dart` without the entrypoint
+const String _envEntryPointHome = 'SIDEKICK_ENTRYPOINT_HOME';
+
+/// Environment variable containing the location of the dart package of this
+/// sidekick CLI. It contains the source code and tool scripts of this sidekick
+/// CLI.
+///
+/// SIDEKICK_PACKAGE_HOME is set by the `tool/run.sh` script and always called
+/// when executed via `entryPoint`
+const String _envPackageHome = 'SIDEKICK_PACKAGE_HOME';
+
+class SidekickContext {
+  SidekickContext._();
+
+  // TODO add context cache that SidekickRunner can inject while executing a
+  //  Command to prevent multiple (expensive) lookups
+  // SidekickContextCache cache;
+
+  /// Returns the name of the CLI
+  static String get cliName {
+    return sidekickPackage.cliName;
+  }
+
+  /// Returns the directory of the Flutter SDK sidekick uses for the [flutter] command
+  ///
+  /// This variable is usually set to a pinned version of the Flutter SDK per project, i.e.
+  /// - https://github.com/passsy/flutter_wrapper
+  /// - https://github.com/fluttertools/fvm
+  ///
+  /// The flutterSdkPath can be set in [initializeSidekick].
+  static Directory? get flutterSdk => throw 'TODO';
+
+  /// Returns the directory of the Dart SDK sidekick uses for the [dart] command
+  ///
+  /// Overrides the Dart SDK of [flutterSdk] when set
+  ///
+  /// The dartSdkPath can be set in [initializeSidekick].
+  static Directory? get dartSdk => throw 'TODO';
+
+  /// The main package/app which should be executed by the [flutter] command
+  ///
+  /// The mainProjectPath can be set in [initializeSidekick].
+  ///
+  /// It's optional, not every project has a mainProject, there are repositories
+  /// with zero or multiple projects.
+  static DartPackage? get mainProject => throw 'TODO';
```

Comment:
@Giuspepe Do you think they belong to `SidekickContext`?

Reply:
Yes I think I would rather have them here than somewhere else again

Reply:
I don't quite think they belong there after refactoring. Anyhow, I wouldn't move them there in this PR